### PR TITLE
Use the actual current period on the usage page

### PR DESCRIPTION
**Pull request by bonysureliya**
## Description

from date = previous month date
to date = current date

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/15544

## Release Notes

```release-note
Use the actual current period when visiting the usage page
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@bonysureliya: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by gtsiolis** [comment]
/werft run with-clean-slate-deployment=true with-preview=true recreate-vm=true

**Review by gtsiolis — CHANGES_REQUESTED** [review]
Thanks for contributing, @bonysureliya! 🍊 

I couldn't test this in the preview environment as the build was failing[[1](https://werft.gitpod-dev.com/job/gitpod-build-patch-1-fork.12)]:

```
Failed to compile.

src/components/UsageView.tsx
Syntax error: 'finally' expected (57:8)
```

1.  Could you fix this—I'll retrigger the build once you push some changes.
2. You'll also need to sign a CLA[[1](https://powerforms.docusign.net/30e1aa46-cce5-4bc6-a933-ad02cd8d5ada?env=eu&acct=86335c2c-01b9-4edc-a187-014f5150a112&accountId=86335c2c-01b9-4edc-a187-014f5150a112)] once before merging your first contribution. Cc @meysholdt

**Comment by gtsiolis** [comment]
@bonysureliya Friendly reminder from https://github.com/gitpod-io/gitpod/pull/16474#pullrequestreview-1305666955:

> You'll also need to sign a CLA[[1](https://powerforms.docusign.net/30e1aa46-cce5-4bc6-a933-ad02cd8d5ada?env=eu&acct=86335c2c-01b9-4edc-a187-014f5150a112&accountId=86335c2c-01b9-4edc-a187-014f5150a112)] once before merging your first contribution. Cc @meysholdt

**Comment by meysholdt** [comment]
Thank you, @bonysureliya, for signing our CLA!

**Comment by gtsiolis** [comment]
@bonysureliya The usage page still defaults to **Month 1st - Today**.

You can check the preview environment below:

> https://patch-1.preview.gitpod-dev.com/workspaces

**Review by gtsiolis — COMMENTED** [review]
@bonysureliya I've added one comment above. This seems to behave similarly as the current state, ses https://github.com/gitpod-io/gitpod/pull/16474#issuecomment-1463679242.

File: components/dashboard/src/components/UsageView.tsx [hunk 1]
```diff
@@ -41,9 +41,14 @@ function UsageView({ attributionId }: UsageViewProps) {
         const match = /#(\d{4}-\d{2}-\d{2}):(\d{4}-\d{2}-\d{2})/.exec(location.hash);
         if (match) {
             try {
-                setStartDate(dayjs(match[1], "YYYY-MM-DD"));
-                setEndDate(dayjs(match[2], "YYYY-MM-DD"));
+                const today = new Date();
+                const prevMonth = new Date(today.getFullYear(), today.getMonth() - 1, today.getDate());
+                const todayStr = today.toISOString().slice(0, 10);
+                const prevMonthStr = prevMonth.toISOString().slice(0, 10);
```
Comment (easyCZ) [inline components/dashboard/src/components/UsageView.tsx:47]: If I'm reading this correctly, this change would ignore the the URL hash parameter and would always set it to one month back. That's actually not the desired behaviour, as it prevents deep-linking of the current view in the URL.

Showing the last 1 month, could be the default behaviour when no hash is supplied, however, but the change as it stands does not do that. You likely want to move the handling into the `!match` case of this if.

**Review by easyCZ — CHANGES_REQUESTED** [review]

**Review by bonysureliya — COMMENTED** [review]
Review of Date Changing Functionality , @easyCZ  Helped where it was catching bug

**Comment by gtsiolis** [comment]
@bonysureliya Is this ready for another review? I'd love to get this merged and fix this annoying issue in https://github.com/gitpod-io/gitpod/pull/16474.

Could you also make sure this is up-to-date with the `main` branch?

Cc @easyCZ

**Comment by bonysureliya** [comment]
Yes I will be completing the issue in one or two days.

**Pull request closed without merging**